packages/connect-node/src/connect-http-transport.ts

```diff
@@ -446,9 +443,25 @@ function makeNodeRequest(options: NodeRequestOptions) {
   });
 }
 
-function nodeRequest(protocol: string) {
-  if (protocol.startsWith("http") || protocol.startsWith("https")) {
-    return protocol.includes("https") ? https.request : http.request;
-  }
-  throw new Error("Unsupported protocol");
+function getNodeRequest(
+  req: StreamingRequest,
+  httpOptions: http.RequestOptions | https.RequestOptions | undefined
+) {
+  return new Promise<http.ClientRequest>((resolve, reject) => {
+    const endpoint = new URL(req.url);
+    const nodeRequestFn = nodeRequest(endpoint.protocol);
+    const request = nodeRequestFn(req.url, {
+      headers: webHeaderToNodeHeaders(req.header),
+      method: "POST",
+      path: endpoint.pathname,
+      signal: req.signal,
+      ...httpOptions,
+    });
+    request.on("socket", (socket) => {
+      socket.on("connect", () => {
+        resolve(request);
+      });
+      socket.on("error", reject);
+    });
```

Comment:
From what I can see from the [documentation](https://nodejs.org/api/http.html#class-httpagent), `http` opens a new socket for every request by default. But it is likely that users may want to provide their own Agent and keep a socket alive across multiple requests. 

We have to support this use case, but we don't have to do it in this PR. Can you take a note?

In this PR, we have to make sure we don't leave too many event listeners attached:
- use `once("socket", ...)`
- for "connect" and "error", `once()` should also be fine, but we have to detach the "connect" listener on "error", and we have to detach the "error" listener on "connect"

This way, there should be any code path where we leave any event listeners attached we don't need.